### ISSUE TITLE
use v-html leads to  'not matching server-rendered content'

### ISSUE THREAD

**Issue by ken1987** [vuejs/vue#3301]
Add the code contains v-html in  [vue-ssr-demo](https://github.com/yyx990803/vue-ssr-demo), the results appear as shown below
````
[Vue warn]: The client-side rendered virtual DOM tree is not matching server-rendered content. Bailing hydration and performing full client-side render. 
````

new_component.js 
```js
// new_component.js 
export default {
  props: ['cols'],
  template:"<div v-html='cols'></div>"
}
```
app.js
````
....
import Nc from './components/new_component'
export default {
  render (h) {
    return (
      <div id="app">
        <Nc cols={store.state.cols} />
        .........
      </div>
    )
  },
  ....
  }
}
````

By a preliminary investigation, vnode.children in new component is empty

**Comment by yyx990803** [comment]
Need more information. What HTML do you have in `cols`?

**Comment by ken1987** [comment]
like the [fork](https://github.com/ken1987/vue-ssr-demo/tree/master/src/components)

**Comment by posva** [comment]
@ken1987 A jsfiddle would be better 😉

**Comment by yyx990803** [comment]
@posva SSR could be hard to repro in jsfiddle, the fork should be fine, I'll take a look when I have time

**Comment by erbing** [comment]
@yyx990803  Will, You must be downloaded pokeman GO  ...

**Comment by erbing** [comment]
@yyx990803  You must be downloaded pokeman GO...

**Comment by Mrxdh** [comment]
@erbing  i like it

**Comment by ken1987** [comment]
A similar problem occurs when a variable is equal to an empty string
````
let MyDom=''
export default {
    render(h) {
        return (
           ....
           {MyDom}
           ....
        )
    }
}
````

**Comment by posva** [comment]
@yyx990803 Indeed, I overlooked that. Sorry for that @ken1987 !

**Comment by yyx990803** [comment]
Fixed via 5f560a4
